exporter/prometheusremotewriteexporter/exporter_test.go

```diff
@@ -727,6 +755,14 @@ func Test_PushMetrics(t *testing.T) {
 						BackOffConfig: retrySettings,
 					}
 
+					if tt.enableSendingRW2 {
+						cfg.RemoteWriteProtoMsg = config.RemoteWriteProtoMsgV2
+						err := featuregate.GlobalRegistry().Set("exporter.prometheusremotewritexporter.enableSendingRW2", true)
```

Comment:
Since this is a global variable, if you enable the feature gate here it will take effect in all subsequent tests.

A pattern I like to use is:

```go
oldValue := enableSendingRW2FeatureGate.IsEnabled()
testutil.SetFeatureGateForTest(t, enableSendingRW2FeatureGate, tt.enableSendingRW2)
defer testutil.SetFeatureGateForTest(t, enableSendingRW2FeatureGate, oldValue)
```

Reply:
fixed.